### PR TITLE
qemu.tests: add case for external snapshot

### PR DESCRIPTION
**Pull request by pingl2016**
1. move function to parent class fo reducing repetitive code.
2. correct the snapshot tag to get the right guest_file_name
for windows guest.
3. add case to change block interface for snapshot

Signed-off-by: Ping Li <pingl@redhat.com>

ID: 1374196, 1374199

### CODE REVIEW

**Comment by aliang123** [comment]
Test on both RHEL.7.3 and Win8.1 pass, so acked.

Acked-by: Aihua Liang <aliang@redhat.com>

**Comment by CongLi** [comment]
Acked-by: Cong Li <coli@redhat.com>

File: qemu/tests/qemu_disk_img.py [hunk 1]
```diff
@@ -176,6 +177,32 @@ def verify_info(self, params=None):
                 msg += "Expect: %s, actual: %s" % (evalue, avalue)
                 raise error.TestFail(msg)
 
+    @error_context.context_aware
+    def check_backingfile(self):
+        error_context.context("check image('%s') backing file" %
+                              self.image_filename, logging.info)
+        out = self.get_info()
```
Comment (xutian) [inline qemu/tests/qemu_disk_img.py]: useless check

File: qemu/tests/qemu_disk_img.py [hunk 2]
```diff
@@ -176,6 +177,32 @@ def verify_info(self, params=None):
...
+        if not out:
+            msg = "Fail to get image('%s') info" % self.image_filename
+            raise error.TestFail(msg)
+        backingfile = re.search(r'backing file: +(.*)', out, re.M)
```
Comment (xutian) [inline qemu/tests/qemu_disk_img.py]: instead with match

Comment (xutian) [inline qemu/tests/qemu_disk_img.py]: if match:
     backingfile = match.group(1)

Comment (xutian) [inline qemu/tests/qemu_disk_img.py]: try:
      backingfile =  re.search(r'backing file: +(.*)', out, re.M).group(1)
      if backingfile != self.base_image_filename:
            raise error.TestFailed(xxxx)
 except TypeError:
      if  not (self.base_tag != "null" and  self.base_tag):
            raise

File: qemu/tests/cfg/qemu_disk_img.cfg [hunk 3]
```diff
@@ -125,14 +125,16 @@
...
+            image_chain += " snA"
+            image_name_snA = "images/snA"
+            image_format_snA = "qcow2"
+            guest_file_name_snA = "${tmp_dir}/snA"
             variants:
                 - @default:
                 - backing_chain:
                     backing_chain = yes
+                - change_block_interface:
+                    drive_format_snA = ide
```
Comment (xutian) [inline qemu/tests/cfg/qemu_disk_img.cfg]:
```
q35:
   ahci
```

**Comment by pingl2016** [comment]
(01/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.qemu_disk_img.info:  PASS (133.77 s)
 (02/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.qemu_disk_img.info.backing_chain:  PASS (94.87 s)
 (03/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.qemu_disk_img.info.change_block_interface:  PASS (115.32 s)
 (04/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snB.to_base:  PASS (230.32 s)
 (05/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snB.to_base.cache_mode.none:  PASS (222.26 s)
 (06/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snB.to_base.cache_mode.writeback:  PASS (223.47 s)
 (07/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snB.to_base.cache_mode.unsafe:  PASS (220.37 s)
 (08/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snB.to_base.cache_mode.directsync:  PASS (224.37 s)
 (09/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snB.to_base.cache_mode.writethrough:  PASS (221.25 s)
 (10/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snB.to_null:  PASS (241.68 s)
 (11/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snC.to_base:  PASS (275.91 s)
 (12/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snC.to_new:  PASS (285.28 s)
 (13/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snC.to_new.restore_to_snB:  PASS (339.21 s)
 (14/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snD.to_snB_snA_base:  PASS (487.81 s)
 (29/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win2016.x86_64.io-github-autotest-qemu.qemu_disk_img.info:  PASS (342.08 s)
 (30/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win2016.x86_64.io-github-autotest-qemu.qemu_disk_img.info.backing_chain:  PASS (194.81 s)
 (31/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win2016.x86_64.io-github-autotest-qemu.qemu_disk_img.info.change_block_interface:  PASS (210.96 s)
 (32/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win2016.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snB.to_base:  PASS (458.24 s)
 (33/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win2016.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snB.to_base.cache_mode.none:  PASS (459.48 s)
 (34/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win2016.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snB.to_base.cache_mode.writeback:  PASS (457.56 s)
 (35/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win2016.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snB.to_base.cache_mode.unsafe:  PASS (467.19 s)
 (36/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win2016.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snB.to_base.cache_mode.directsync:  PASS (461.86 s)
 (37/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win2016.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snB.to_base.cache_mode.writethrough:  PASS (470.65 s)
 (38/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win2016.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snB.to_null:  PASS (512.59 s)
 (39/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win2016.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snC.to_base:  PASS (563.78 s)
 (40/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win2016.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snC.to_new:  PASS (628.53 s)
 (41/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win2016.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snC.to_new.restore_to_snB:  PASS (713.88 s)
 (42/84) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.Win2016.x86_64.io-github-autotest-qemu.qemu_disk_img.rebase.snD.to_snB_snA_base:  PASS (1014.25 s)

File: qemu/tests/qemu_disk_img.py [hunk 4]
```diff
@@ -176,6 +178,27 @@ def verify_info(self, params=None):
...
 
+    @error_context.context_aware
+    def check_backingfile(self):
+        error_context.context("check image('%s') backing file" %
+                              self.image_filename, logging.info)
+        out = self.get_info()
+        try:
+            backingfile = re.search(r'backing file: +(.*)', out, re.M).group(1)
+            if not self.base_tag or self.base_tag == "null":
+                msg = ("Expected backing file is null")
```
Comment (xutian) [inline qemu/tests/qemu_disk_img.py]: move content from line#199 to line#195 behind except.

**Comment by xutian** [comment]
LGTM,

Acked-by: Xu Tian <xutian@redhat.com>

**Pull request merged**